### PR TITLE
Issue-40: Stage files to s3 destination

### PR DESCRIPTION
**Pull request by eigenbeam**

### CODE REVIEW

File: src/nsidc/metgen/metgen.py [hunk 1]
```diff
@@ -73,6 +79,7 @@ def init_config(configuration_file):
     cfg_parser.set(DESTINATION_SECTION_NAME, "local_output_dir", Prompt.ask("Local output directory", default="output"))
     cfg_parser.set(DESTINATION_SECTION_NAME, "ummg_dir", Prompt.ask("Local UMM-G output directory (relative to local output directory)", default="ummg"))
     cfg_parser.set(DESTINATION_SECTION_NAME, "kinesis_stream_name", Prompt.ask("Kinesis stream name"))
+    cfg_parser.set(DESTINATION_SECTION_NAME, "staging_bucket_name", Prompt.ask("Cumulus s3 bucket name"))
```
Comment (juliacollins) [inline src/nsidc/metgen/metgen.py:82]: The value previously used in the template in the function s3_url (`s3://nsidc-cumulus-${environment}-ingest-staging/external/${auth_id}/${version}/${uuid}/${s3_name}`) is the "official" s3 url pattern. It makes sense to have a means of overriding the default, but my understanding is `s3://nsidc-cumulus-${environment}-ingest-staging/external/` should be the default value for the root of the staging bucket name.

File: src/nsidc/metgen/metgen.py [hunk 2]
```diff
@@ -289,10 +294,19 @@ def checksum(file):
     return sha256.hexdigest()
 
 def s3_url(mapping, name):
-    mapping['s3_name'] = name
-    template = Template('s3://nsidc-cumulus-${environment}-ingest-staging/external/${auth_id}/${version}/${uuid}/${s3_name}')
+    """
```
Comment (juliacollins) [inline src/nsidc/metgen/metgen.py:297]: See comment above. We don't want to make Ops users type in the default value when creating an ini file.

File: src/nsidc/metgen/metgen.py (hunk 1 above)
Comment (juliacollins) [reply src/nsidc/metgen/metgen.py:82]: Based on the documentation for CNM messages, we'd want to flag any CNM content that didn't include the default pattern `s3://nsidc-cumulus-${environment}-ingest-staging/external/${auth_id}/${version}/${uuid}/${s3_name}` as invalid. We want to be able to test using non-official staging buckets, though, so this is more food for thought regarding the meaning of "valid" when it comes to CNM messages.

**Review by juliacollins — CHANGES_REQUESTED** [review]
We'll need a default value for the s3 url which matches the official s3 url content and structure, so Ops users don't need to do anything special to use the correct s3 staging url.

File: src/nsidc/metgen/metgen.py (hunk 1 above)
Comment (eigenbeam) [reply src/nsidc/metgen/metgen.py]: I've reworked the default values, added tests, and set the default bucket & kinesis stream names to the cumulus patterns.

File: src/nsidc/metgen/metgen.py (hunk 2 above)
Comment (eigenbeam) [reply src/nsidc/metgen/metgen.py:286]: I've reworked the default values, added tests, and set the default bucket & kinesis stream names to the cumulus patterns.

**Review by juliacollins — APPROVED** [review]

**Pull request merged**